src/commands/build.ts

```diff
@@ -225,11 +225,24 @@ export async function command(commandOptions: CommandOptions) {
         // We treat ".proxy.js" files special: we need to make sure that they exist on disk
         // in the final build, so we mark them to be written to disk at the next step.
         if (resolvedImportUrl.endsWith('.proxy.js')) {
-          allProxiedFiles.add(
-            resolvedImportUrl.startsWith('/')
-              ? path.resolve(cwd, spec)
-              : path.resolve(path.dirname(outLoc), spec),
-          );
+          // handle proxied files from web_modules
+          if (resolvedImportUrl.startsWith(config._webModulesPath)) {
+            allProxiedFiles.add(
+              path.resolve(
+                cwd,
+                config.devOptions.out,
+                config._webModulesPath.replace(/^\//, ''),
+                spec,
+              ),
+            );
+          } else {
+            // handle local proxied files
+            allProxiedFiles.add(
+              resolvedImportUrl.startsWith('/')
+                ? path.resolve(cwd, spec)
+                : path.resolve(path.dirname(outLoc), spec),
+            );
+          }
```

Comment:
This feels dumb

How can we un-dumb this?

Reply:
Looks like this is the solution for https://github.com/pikapkg/snowpack/issues/549

Yea, the idea of adding new files or making changes in web_modules/ directory after an install feels wrong. I don't have a better solution... but yea. Do you mind pulling this out and into a separate PR so that we don't block the `tippy.js` issue?

Reply:
> Do you mind pulling this out and into a separate PR so that we don't block the tippy.js issue?

Oh no I more meant this line: `path.resolve(cwd, config.devOptions.out, config._webModulesPath.replace(/^\//, ''), spec)`. Is there a better way to resolve this?

This actually does nothing to the installed files; we’re just simply reading them wheras before we were trying to find them in your source built directory.

Reply:
Also the tippy.js test won’t pass without this—this is a larger issue where we aren’t correctly proxying files in web_modules at all.

Reply:
Oh but yes I believe this does fix #549 as well

Reply:
Let me think about this a bit then. If I'm reading this right, what this change is telling Snowpack to do is "add a .proxy.js file into the web_modules directory after the install has run". That's the normal logic for proxy modules, but it feels wrong to inject files into the web_modules/ directory like this. In my mind, that's a sacred directory owned completely by the install step.

Reply:
Yeah, good point. Well I think this is still the right starting point, because it was failing at the _read_ step, not the _write_ step. The write step would be more involved from here.

I think the fact that it writes to the web_modules directory is just how it naturally functions. So we’d need to override that behavior below when writing.

Reply:
Thought about this a little more and it seems like there are 2 options to me?

**Generate files in `web_modules`**
The solution here 👆. To your point we’ve lost our “purity” of that package because we’ve put our own files in there. But on the other hand, those are files for this package and this package alone, so this feels like the most logical place to put them (esp. for caching)

**Create some new `web_modules_meta` dir**
Ignore the name, but we have some concept of a meta directory that contains the generated proxy files. This would basically have to duplicate the folder structure of `web_modules`, because they are meant for those modules. We’d need this because there’s no logical place to put them in the source directory—what if 3 files use the same CSS file? We shouldn’t duplicate that 3 times next to those files; there should be some global ref for them.

Reply:
Yea, having slept on it a bit I'm fine with adding the proxy files to web_modules.